src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs

```diff
@@ -29,14 +29,14 @@ public class WebFunctionsManager : IWebFunctionsManager
         private const string DurableTaskStorageConnectionName = "azureStorageConnectionStringName";
         private const string DurableTask = "durableTask";
 
-        private readonly ScriptJobHostOptions _hostOptions;
+        private readonly IOptionsMonitor<ScriptApplicationHostOptions> _applicationHostOptions;
         private readonly ILogger _logger;
         private readonly HttpClient _client;
         private readonly IEnumerable<WorkerConfig> _workerConfigs;
 
-        public WebFunctionsManager(IOptions<ScriptApplicationHostOptions> applicationHostOptions, IOptions<LanguageWorkerOptions> languageWorkerOptions, ILoggerFactory loggerFactory, HttpClient client)
+        public WebFunctionsManager(IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions, IOptions<LanguageWorkerOptions> languageWorkerOptions, ILoggerFactory loggerFactory, HttpClient client)
```

Comment:
Use of OptionsMonitor ensures that we can always access the current fresh value, which is modified during specialization. This is a WebHost level service so lives on through specialization, so it can't cache values on construction.

Reply:
This isn't really SeaBreeze specific - I wonder why we haven't seen this on Windows? Perhaps in the windows case we don't let any requests hit the site until DWAS specializes it, ensuring that teh WebFunctionManager isn't loaded pre-specialization and can't cache placeholder values. In the case of SeaBreeze it looks like requests can come in before the site is specialized, right @balag0 ?

Reply:
Yes. With linux the request doesn't wait for specialization to complete. We start the specialization and the request can come in right away. 

We rely on EnvironmentReadyCheckMiddleware to block the requests until specialization is complete.